partiql-parser/src/lalr/lexer.rs

```diff
@@ -447,11 +538,34 @@ mod tests {
                               */
                       'b':1}` "#;
         let mut lexer = Lexer::new(ion_value);
-        if let (_start, Token::Ion(s), _end) = lexer.next().unwrap().unwrap() {
-            assert_eq!(ion_value.trim().trim_matches('`'), s);
-        } else {
-            panic!("Lex failed")
-        }
+
+        let tok = lexer.next().unwrap().unwrap();
+        assert!(matches!(tok, (1, Token::Ion(ion_str), 154) if ion_str == ion_value.trim()));
+
+        assert_eq!(
+            EmbeddedIonLexer::new(ion_value.trim()).into_iter().count(),
+            1
+        );
+    }
+
+    #[test]
+    fn nested_comments() {
+        let comments = r##"/*  
+                                    /*  / * * * /
+                                    /*  ' " ''' ` 
+                                    */  text
+                                    */  1 2 3 4 5 6,7,8,9 10.112^5
+                                    */"##;
+
+        let nested_lex = CommentLexer::new(comments).with_nesting();
+        assert_eq!(nested_lex.into_iter().count(), 1);
```

Comment:
qq: just trying to understand how iterator works in this case.

Here we have 3 blocks nested inside each other:
```bash
/*
 /*
   / * * * /
 

   /*
     ' " ''' ` 
   */

   text
  */

  1 2 3 4 5 6,7,8,9 10.112^5
*/
```

Why does the _count_ returns 1?

Reply:
Lexers really aren't designed to deal with nested or recursive structures; that's the domain of parsers. Here, the idea is to parse this as 1 single comment (the outer comment). It happens to contain other comments, in the same way that a string may contain escaped quotes.

At the end of the day, we really just want to skip past comments, so recognizing the nesting and encompassing it is sufficient.

Reply:
> Lexers really aren't designed to deal with nested or recursive structures; that's the domain of parsers. Here, the idea is to parse this as 1 single comment (the outer comment). It happens to contain other comments, in the same way that a string may contain escaped quotes.
> 
> At the end of the day, we really just want to skip past comments, so recognizing the nesting and encompassing it is sufficient.

Makes sense, thanks for elaborating. Totally resonate with that (we can't use FAs for nested structures).